### DISCUSSION TITLE
Add a slider velocity progression tool for the osu!taiko editor

### DISCUSSION THREAD

**Discussion by Hiviexd** [ppy/osu#24876] [Ideas]
first of all, i'd like to emphasize on how important this kind of tool is, it has practically shaped the modern mapping meta, and is basically  a prerequisite for mapping anything that involves some kind of SV that's not an abrupt change. this tool is so influential that almost any active mapper would tell you that they've used it at a recent point in time, and its creator has earned a community contributors recognition from it ([this newspost](https://osu.ppy.sh/home/news/2022-06-30-community-contributors-2021), `ctrl + F "frukoyurdakul"`)

this tool creates SV speedups and slowdowns on a given set of objects (notes, drumrolls, and swells) using a specific formula, with linear being the most common one.

**Proposed workflow:**
- user would select a set of objects in the timeline
- in the right click menu, there'd be an option for this SV tool
- clicking the option would reveal a pop-up that has the following fields
  - starting SV (SV of the first note)
  - target SV (SV of the last note)
  - formula (**linear**, exponential, ease-in, ease-out, ease-in-out)
  - an "apply" button
 - after filling in the fields, the tool would place SVs on every selected object in order to satisfy the conditions we gave it

i'm happy to answer any questions about this, as i genuinely believe that having this tool in lazer would give me and many other mappers a very good reason to actually switch from stable to lazer for mapping.

**Comment by peppy** [comment]
There's a plan to add tweening effect sections eventually, which is what is being asked for here.
